zerver/management/commands/process_queue.py

```diff
@@ -35,6 +37,18 @@ def handle(self, *args, **options):
         logging.basicConfig()
         logger = logging.getLogger('process_queue')
 
+        def restart_senders_and_exit_with_three(signal, frame):
+            # type: (int, FrameType) -> None
+            """
+            This process is watched by Django's autoreload, so exiting
+            with status code 3 will cause this process to restart.
+            """
+            logger.warn("Signal(10) received. Restarting the queue processors.")
+            if settings.PRODUCTION:
+                subprocess.check_call(["supervisorctl", "restart", "zulip-senders:*"])
```

Comment:
Why do you have this restart operation here?  I would think it would be unnecessary.